File: BringMeHome/BringMeHome.Services/Database/City.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using static BringMeHome.Services.Database.Adopters;

namespace BringMeHome.Services.Database
{
    public class City
    {
        [Key]
        public int CityID { get; set; }

        [Required, StringLength(100)]
        public string CityName { get; set; }

        public int CantonID { get; set; }

        // Navigation properties
        [ForeignKey("CantonID")]
        public virtual Canton Canton { get; set; }

        public virtual ICollection<Shelter> Shelters { get; set; }
        public virtual ICollection<Adopter> Adopters { get; set; }
    }
}

File: bringmehomeUI/bringmehome_admin/lib/models/city.dart
class City {
  final int? cityID;
  final String cityName;

  City({this.cityID, required this.cityName});

  factory City.fromJson(Map<String, dynamic> json) {
    return City(
      cityID: json['cityID'] as int?,
      cityName: json['cityName'] as String,
    );
  }
  Map<String, dynamic> toJson() {
    return {
      'cityID': cityID,
      'cityName': cityName,
    };
  }
}
